### ISSUE TITLE
False positive when using ndarray.reshape with separate arguments for each element of the shape

### ISSUE THREAD

**Issue by ntraut** [PyCQA/pylint#7883]
### Bug description

```
# pylint: disable=missing-docstring
x = np.zeros(12)
x.reshape(3, 2, 2, order='F')
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module src.a
a.py:4:0: E1121: Too many positional arguments for method call (too-many-function-args)
a.py:4:0: E1124: Argument 'order' passed by position and keyword in method call (redundant-keyword-arg)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

no error reported, the syntax is correct as documented [here](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.reshape.html#numpy-ndarray-reshape)

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.9.13 | packaged by conda-forge | (main, May 27 2022, 17:00:33) 
[Clang 13.0.1 ]
```

### OS / Environment

macOS 12.6

### Additional dependencies

numpy @ file:///private/var/folders/.../croots/recipe/numpy_and_numpy_base_1652801688716/work

**Comment by zhcui** [comment]
Any update on this? Thanks. The false positive still exists.

**Comment by Pierre-Sassoulas** [comment]
Hello @zhcui as you can see, there's a merge request opened in astroid : https://github.com/pylint-dev/astroid/pull/1999. If you want to take it over it would probably speed up the resolution.
